custom_components/rfplayer/rflib/rfpparser.py

```diff
@@ -79,15 +104,17 @@ def decode_packet(packet: str) -> list:
         data["message"] = packet.replace("ZIA--", "")
         return [data]
 
-    # Protocols
-    message = json.loads(packet.replace("ZIA33", ""))["frame"]
+ 
+    if packet.startswith("ZIA33"):
+        message = json.loads(packet.replace("ZIA33", ""))["frame"]
```

Comment:
si message n'est défini que dans ce `if`, il va y avoir un soucis ligne 93 par exemple

Reply:
avant, la varialbe `message` était défini à chaque appel de `decode_packet`, en mettant cette ligne dans le `if packet.startswith("ZIA33"):`, `message` n'est plus défini si le packet ne commence plus par `ZIA33`.
donc quand on va rentrer dans `data["protocol"] in ["X2D"]:`, au moment où il va vouloir faire `if message["infos"]["subTypeMeaning"] == 'Detector/Sensor':`, il y aura une erreur

Reply:
ok j'avais pas vu que la ligne était déplacée ...